scripts/data/utreexo.py

```diff
@@ -0,0 +1,121 @@
+from poseidon_py.poseidon_hash import poseidon_hash_many
+
+# The array of trees in the forest
+# [T_1, T_2, T_4, T_8, ... ]
+root_nodes = [None] * 27
+
+# The set of leaf nodes in the forest
+leaf_nodes = dict()
+
+
+class Node:
+    def __init__(self, key, left=None, right=None):
+        self.val = key
+        self.left = left
+        self.right = right
+        self.parent = None
+
+
+def parent_node(root1, root2):
```

Comment:
@m-kus In this version, we save all data using the Node type.
But in our struct we only store hashes:
pub struct UtreexoState {
    pub roots: Span<Option<felt252>>,
    pub num_leaves: u64,
}

Do we want the full tree? Or will there be a full/light utreexo system?

Reply:
`Node` contains extra pointers that simplify traversing the tree.
We don't need this in the Cairo implementation.